classes/Router/RoutesConfig.php

```diff
@@ -0,0 +1,199 @@
+use PrestaShop\Module\AutoUpgrade\Controller\ErrorReportController;
+use PrestaShop\Module\AutoUpgrade\Controller\HomePageController;
+use PrestaShop\Module\AutoUpgrade\Controller\LogsController;
+use PrestaShop\Module\AutoUpgrade\Controller\RestorePageBackupSelectionController;
+use PrestaShop\Module\AutoUpgrade\Controller\RestorePagePostRestoreController;
+use PrestaShop\Module\AutoUpgrade\Controller\RestorePageRestoreController;
+use PrestaShop\Module\AutoUpgrade\Controller\UpdatePageBackupController;
+use PrestaShop\Module\AutoUpgrade\Controller\UpdatePageBackupOptionsController;
+use PrestaShop\Module\AutoUpgrade\Controller\UpdatePagePostUpdateController;
+use PrestaShop\Module\AutoUpgrade\Controller\UpdatePageUpdateController;
+use PrestaShop\Module\AutoUpgrade\Controller\UpdatePageUpdateOptionsController;
+use PrestaShop\Module\AutoUpgrade\Controller\UpdatePageVersionChoiceController;
+
+class RoutesConfig
+{
+    const ROUTES =
+        /* HOME PAGE */
+        [
+            Routes::HOME_PAGE => [
+                'controller' => HomePageController::class,
+                'method' => 'index',
+            ],
+            Routes::HOME_PAGE_SUBMIT_FORM => [
+                'controller' => HomePageController::class,
+                'method' => 'submit',
+            ],
+            /* UPDATE PAGE */
+            /* step: version choice */
+            Routes::UPDATE_PAGE_VERSION_CHOICE => [
+                'controller' => UpdatePageVersionChoiceController::class,
+                'method' => 'index',
+            ],
+            Routes::UPDATE_STEP_VERSION_CHOICE => [
+                'controller' => UpdatePageVersionChoiceController::class,
+                'method' => 'step',
+            ],
+            Routes::UPDATE_STEP_VERSION_CHOICE_SAVE_FORM => [
+                'controller' => UpdatePageVersionChoiceController::class,
+                'method' => 'save',
+            ],
+            Routes::UPDATE_STEP_VERSION_CHOICE_SUBMIT_FORM => [
+                'controller' => UpdatePageVersionChoiceController::class,
+                'method' => 'submit',
+            ],
+            Routes::UPDATE_STEP_VERSION_CHOICE_CORE_TEMPERED_FILES_DIALOG => [
+                'controller' => UpdatePageVersionChoiceController::class,
+                'method' => 'coreTemperedFilesDialog',
+            ],
+            Routes::UPDATE_STEP_VERSION_CHOICE_THEME_TEMPERED_FILES_DIALOG => [
+                'controller' => UpdatePageVersionChoiceController::class,
+                'method' => 'themeTemperedFilesDialog',
+            ],
+            /* step: update options */
+            Routes::UPDATE_PAGE_UPDATE_OPTIONS => [
+                'controller' => UpdatePageUpdateOptionsController::class,
+                'method' => 'index',
+                'middleware' => [
+                    RoutesMiddleware::UPDATE_IS_CONFIGURED,
+                    RoutesMiddleware::LOCAL_CHANNEL_XML_AND_ZIP_EXIST,
```

Comment:
Is having the whole definition of `RoutesMiddleware` necessary? Looks like the whole system would be simpler by having directly the classes here.

```suggestion
                    UpdateIsConfigured::class,
                    LocalChannelXmlAndZipExist::class,
```